File: eco-api/src/main/java/com/willfp/eco/util/ServerUtils.java
package com.willfp.eco.util;

import com.willfp.eco.core.Eco;

/**
 * Utilities / API methods for the server.
 */
public final class ServerUtils {
    /**
     * Get the current server TPS.
     * <p>
     * The value is capped at 20, so a server running above nominal speed still reports 20.
     *
     * @return The TPS, at most 20.
     */
    public static double getTps() {
        double tps = Eco.get().getTPS();

        if (tps > 20) {
            return 20;
        } else {
            return tps;
        }
    }

    private ServerUtils() {
        throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
    }
}

File: eco-api/src/main/kotlin/com/willfp/eco/util/ServerUtils.kt
@file:JvmName("ServerUtilsExtensions")

package com.willfp.eco.util

import org.bukkit.Server

/**
 * The current TPS (ticks per second) of the running server, clamped to a maximum of 20.
 *
 * The receiver is ignored; this always reports the TPS of the server this plugin is running on.
 *
 * @see ServerUtils.getTps
 */
val Server.tps: Double
    get() = ServerUtils.getTps()
